superbench/benchmarks/micro_benchmarks/computation_communication_overlap.py

```diff
@@ -0,0 +1,255 @@
+            self.__local_rank = int(os.environ['LOCAL_RANK'])
+            # if self.__world_size < 2:
+            # raise Exception("Distributed env check error, WORLD_SIZE < 2")
+        except BaseException as e:
+            self._result.set_return_code(ReturnCode.DISTRIBUTED_SETTING_INIT_FAILURE)
+            torch.distributed.destroy_process_group()
+            logger.error('Initialize distributed env failed - benchmark: {}, message: {}.'.format(self._name, str(e)))
+            return False
+
+        if torch.cuda.is_available():
+            torch.cuda.set_device(self.__local_rank)
+
+        return True
+
+    def __kernel_nccl_pipeline(self, kernel, matA, matB, stages, message, times):
+        """Computation and NCCL kernel pipeline with single GPU.
+
+        Args:
+            kernel (ComputationKernelType): the type of the computation kernel to run.
+            matA (list[tensor]): the matrix list used in matmul or mul for every stage.
+            matB (tensor): the matrix used in matmul.
+            stages (int): the ratio number of computation kernel and communication kernel.
+            message(tensor): the data used to be transferred through NCCL.
+            times(int): number of times in one step to run.
+
+        Return:
+            True of False: if computation kernel type is invalid, return False, else, return True.
+        """
+        if kernel == ComputationKernelType.MUL:
+            for i in range(times):
+                torch.distributed.all_reduce(message, op=torch.distributed.ReduceOp.SUM, async_op=True)
+                for stage in range(stages):
+                    matA[stage].mul(matA[stage])
+        elif kernel == ComputationKernelType.MATMUL:
+            for i in range(times):
+                torch.distributed.all_reduce(message, op=torch.distributed.ReduceOp.SUM, async_op=True)
+                for stage in range(stages):
+                    matA[stage].matmul(matB)
+        else:
+            logger.error('Unknown comoputation kernel type - benchmark: {}, type: {}.'.format(self._name, kernel))
+            return False
+        return True
+
+    def _benchmark(self):
+        """Implementation for benchmarking."""
+        M = self._args.m
+        K = self._args.k
+        N = self._args.n
+        P = self._args.p
+        Q = self._args.q
+        kernels = self._args.kernel
+        if self.__local_rank == 0:
+            logger.info(
+                'Computation Communication Overlap - using {} GPUs,\
+                matrix shape for computation: M={} K={} N={},\
+                message tensor shape of NCCL = [{},{}],\
+                ratio between computation kernel and NCCL kernel={}'.format(
+                    self.__world_size, M, K, N, P, Q, self._args.ratio
+                )
+            )
+
+        MatA = list()
+        MatB = list()
+        # Matrix A
+        for _ in range(self._args.ratio):
+            MatA.append(torch.randn(M, K).cuda())
```

Comment:
Do we handle OOM issue if M,K,N are too large?

Reply:
It will be handled in executor outside